src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java

```diff
@@ -97,6 +97,22 @@ public void addListenerForFile(Path file, FileUpdateListener listener) throws IO
         }
     }
 
+    /**
+     * Add a new directory to monitor.
+     *
+     * @param directory The directory to monitor.
+     * @throws IOException if the directory does not exist.
+     */
+    public void addListenerForDirectory(Path directory, FileUpdateListener listener) throws IOException {
+        // This function was created because it makes more sense to call addListenerForDirectory when
+        // the Path is a directory and not a file, even though it does the same thing as addListenerForFile.
+        // The function addListenerForFile works for directories as well. We have to listen to the parent
+        // directory in this case too otherwise if a new file is created in the child directory, no
```

Comment:
This is wrong. You really need to update the DefaultFileMonitorImplementation.

Here a test case that notifications for new files works:

```diff
diff --git a/src/main/java/org/jabref/gui/JabRefExecutorService.java b/src/main/java/org/jabref/gui/JabRefExecutorService.java
index 90c3b31bda..74eb0bbea5 100644
--- a/src/main/java/org/jabref/gui/JabRefExecutorService.java
+++ b/src/main/java/org/jabref/gui/JabRefExecutorService.java
@@ -184,12 +184,12 @@ public class JabRefExecutorService {
         try {
             // This is non-blocking. See https://stackoverflow.com/a/57383461/873282.
             executorService.shutdown();
-            if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
-                LOGGER.debug("One minute passed, {} still not completed. Trying forced shutdown.", executorService.toString());
+            if (!executorService.awaitTermination(10, TimeUnit.SECONDS)) {
+                LOGGER.debug("10 seconds, {} still not completed. Trying forced shutdown.", executorService.toString());
                 // those threads will be interrupted in their current task
                 executorService.shutdownNow();
-                if (executorService.awaitTermination(60, TimeUnit.SECONDS)) {
-                    LOGGER.debug("One minute passed again - forced shutdown of {} worked.", executorService.toString());
+                if (executorService.awaitTermination(10, TimeUnit.SECONDS)) {
+                    LOGGER.debug("10 seconds passed again - forced shutdown of {} worked.", executorService.toString());
                 } else {
                     LOGGER.error("{} did not terminate", executorService.toString());
                 }
diff --git a/src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java b/src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java
index 0090539a96..bcb88189ef 100644
--- a/src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java
+++ b/src/main/java/org/jabref/gui/util/DefaultFileUpdateMonitor.java
@@ -8,6 +8,8 @@ import java.nio.file.StandardWatchEventKinds;
 import java.nio.file.WatchEvent;
 import java.nio.file.WatchKey;
 import java.nio.file.WatchService;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.Optional;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
@@ -32,8 +34,8 @@ public class DefaultFileUpdateMonitor implements Runnable, FileUpdateMonitor {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DefaultFileUpdateMonitor.class);
 
-    private final Multimap<Path, FileUpdateListener> listeners = ArrayListMultimap.create(20, 4);
-    private volatile WatchService watcher;
+    volatile WatchService watcher;
+    final Multimap<Path, FileUpdateListener> listeners = ArrayListMultimap.create(20, 4);
     private final AtomicBoolean notShutdown = new AtomicBoolean(true);
     private final AtomicReference<Optional<JabRefException>> filesystemMonitorFailure = new AtomicReference<>(Optional.empty());
 
@@ -82,7 +84,12 @@ public class DefaultFileUpdateMonitor implements Runnable, FileUpdateMonitor {
     }
 
     private void notifyAboutChange(Path path) {
-        listeners.get(path).forEach(FileUpdateListener::fileUpdated);
+        Collection<FileUpdateListener> fileUpdateListeners = Collections.emptyList();
+        while ((path != null) && (fileUpdateListeners.isEmpty())) {
+            fileUpdateListeners = listeners.get(path);
+            path = path.getParent();
+        }
+        fileUpdateListeners.forEach(FileUpdateListener::fileUpdated);
     }
 
     @Override
@@ -126,6 +133,7 @@ public class DefaultFileUpdateMonitor implements Runnable, FileUpdateMonitor {
             if (watcher != null) {
                 watcher.close();
             }
+            Thread.currentThread().interrupt();
         } catch (IOException e) {
             LOGGER.error("error closing watcher", e);
         }
diff --git a/src/test/java/org/jabref/gui/util/DefaultFileUpdateMonitorTest.java b/src/test/java/org/jabref/gui/util/DefaultFileUpdateMonitorTest.java
new file mode 100644
index 0000000000..95c6235738
--- /dev/null
+++ b/src/test/java/org/jabref/gui/util/DefaultFileUpdateMonitorTest.java
@@ -0,0 +1,40 @@
+package org.jabref.gui.util;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.jabref.gui.JabRefExecutorService;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class DefaultFileUpdateMonitorTest {
+
+    @Test
+    void demonstrateListenerForDirectory(@TempDir Path rootPath) throws Exception {
+        DefaultFileUpdateMonitor fileUpdateMonitor = new DefaultFileUpdateMonitor();
+        JabRefExecutorService.INSTANCE.executeInterruptableTask(fileUpdateMonitor, "FileUpdateMonitor");
+        while (fileUpdateMonitor.watcher == null) {
+            Thread.sleep(100);
+            Thread.yield();
+        }
+        rootPath.register(fileUpdateMonitor.watcher, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
+        final AtomicBoolean called = new AtomicBoolean(false);
+        fileUpdateMonitor.listeners.put(rootPath, () -> {
+            called.set(true);
+        });
+        Files.createFile(rootPath.resolve("test.txt"));
+        int callCount = 0;
+        while ((callCount < 10) && (!called.get())) {
+            Thread.sleep(100);
+            Thread.yield();
+            callCount++;
+        }
+        JabRefExecutorService.INSTANCE.shutdownEverything();
+        assertTrue(called.get());
+    }
+}
```

Maybe, you should add the `Path` of the changed file to `FileUpdateListener`. 

Other approach: implement other classes handling directories and ensure that `DefaultFileUpdateMonitor` handles files only.